x/plans/keeper/plan.go

```diff
@@ -16,37 +16,33 @@ func (k Keeper) AddPlan(ctx sdk.Context, planToAdd types.Plan) error {
 	err := k.plansFs.AppendEntry(ctx, planToAdd.GetIndex(), planToAdd.Block, &planToAdd)
 	if err != nil {
 		details := map[string]string{"planToAdd": planToAdd.String()}
-		return utils.LavaError(ctx, k.Logger(ctx), "AddPlan_add_fixated_entry_failed", details, "could not add new plan fixated entry to storage")
+		return utils.LavaError(ctx, k.Logger(ctx), "AddPlan_add_fixated_entry_failed", details, err.Error())
 	}
 
 	return nil
 }
 
-// GetPlan gets a plan from the KVStore. It increases the plan's refCount by 1
+// GetPlan gets the latest plan from the KVStore and increments its refcount
 func (k Keeper) GetPlan(ctx sdk.Context, index string) (val types.Plan, found bool) {
 	var plan types.Plan
-	err, _ := k.plansFs.GetEntry(ctx, index, &plan)
-	if err != nil {
+	if found := k.plansFs.GetEntry(ctx, index, &plan); !found {
 		return types.Plan{}, false
 	}
 	return plan, true
 }
 
-// FindPlan gets a plan from the KVStore. It does nothing to the plan's refCount
+// FindPlan gets a plan with nearest-smaller block (without changing its refcount)
 func (k Keeper) FindPlan(ctx sdk.Context, index string, block uint64) (val types.Plan, found bool) {
 	var plan types.Plan
-	err, _ := k.plansFs.FindEntry(ctx, index, block, &plan)
-	if err != nil {
+	if found := k.plansFs.FindEntry(ctx, index, block, &plan); !found {
 		return types.Plan{}, false
 	}
 	return plan, true
 }
 
-// PutPlan gets a plan from the KVStore. It decreases the plan's refCount by 1
-func (k Keeper) PutPlan(ctx sdk.Context, index string, block uint64) bool {
-	var plan types.Plan
-	_, found := k.plansFs.PutEntry(ctx, index, block, &plan)
-	return found
+// PutPlan finds a plan with nearest-smaller block and decrements its refcount
+func (k Keeper) PutPlan(ctx sdk.Context, index string, block uint64) {
+	k.plansFs.PutEntry(ctx, index, block)
```

Comment:
handle error

Reply:
`PutEntry()` cannot fail (though it may panic).
Likewise, `PutPlan()` is expected to be called by the holder of a reference. Therefore, it is a serious bug if the Plan is not found.